Google.Api.Generator.Rest/Models/MethodModel.cs

```diff
@@ -15,25 +15,84 @@
 using Google.Api.Generator.Utils;
 using Google.Api.Generator.Utils.Roslyn;
 using Google.Apis.Discovery.v1.Data;
+using Google.Apis.Services;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Collections.Generic;
+using System.Linq;
 using static Google.Api.Generator.Utils.Roslyn.RoslynBuilder;
 
 namespace Google.Api.Generator.Rest.Models
 {
     public class MethodModel
     {
+        private RestMethod _restMethod;
+
         public PackageModel Package { get; }
+        /// <summary>
+        /// Resource that owns this method, if any.
+        /// </summary>
+        public ResourceModel Resource { get; }
+        
+        public IReadOnlyList<ParameterModel> Parameters { get; }
+        public Typ ParentTyp { get; }
+        public Typ RequestTyp { get; }
         public string Name { get; }
 
-        public MethodModel(PackageModel package, string name, RestMethod restMethod)
+        public MethodModel(PackageModel package, ResourceModel resource, string name, RestMethod restMethod)
         {
             Package = package;
+            Resource = resource;
             Name = name.ToUpperCamelCase();
+            ParentTyp = resource?.Typ ?? package.ServiceTyp;
+            RequestTyp = Typ.Nested(ParentTyp, $"{Name}Request");
+            Parameters = CreateParameterList(package, restMethod);
+            _restMethod = restMethod;
+        }
+
+        private static IReadOnlyList<ParameterModel> CreateParameterList(PackageModel package, RestMethod restMethod)
+        {
+            var parameterOrder = restMethod.ParameterOrder ?? new List<string>();
+            // TODO: Skip the Alt parameter
+            // TODO: Handle the body parameter if restMethod.Request is non-null
+            return (restMethod.Parameters ?? new Dictionary<string, JsonSchema>())
+                .Select(pair => new ParameterModel(pair.Key, pair.Value))
+                .OrderBy(p => !p.IsRequired)
```

Comment:
I think you can directly skip here the ones that are not required. Or are you planning to be able to generate overloads later?

Reply:
Not overloads, but they become properties on the request class - we'll definitely want them.